internal/pkg/stream/cfnstack.go

```diff
@@ -130,9 +128,9 @@ func (s *StackStreamer) Fetch() (next time.Time, err error) {
 			// Check for throttles and wait to try again using the StackStreamer's interval.
 			if request.IsErrorThrottle(err) {
 				s.retries += 1
-				return nextFetchDate(s.clock, s.rand, s.retries), nil
+				return nextFetchDate(s.clock, s.rand, s.retries), done, nil
```

Comment:
nit: maybe just explicitly return false, just helps me know that `done` hasn't been modified somewhere up above and you're returning its value (same L133)